pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +76,83 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+//
+// Precedence of failures:
+// 1. A build config's input points to an image stream that does not exist
+// 2. A build config's input uses an image stream tag reference in an existing image stream, but no images within the image stream have that tag assigned
+// 3. A build config's input uses an image stream image reference in an exisiting image stream, but no images within the image stream have the supplied image hexadecimal ID
+func FindMissingInputImageStreams(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, bcInputNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
+			switch bcInputNode.(type) {
+			case *imagegraph.ImageStreamTagNode:
+
+				for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(bcInputNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+					imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+					// note, BuildConfig.Spec.BuildSpec.Strategy.[Docker|Source|Custom]Stragegy.From Input of ImageStream has been converted to ImageStreamTag on the vX to api conversion
+					// prior to our reaching this point in the code; so there is not need to check for that type vs. ImageStreamTag or ImageStreamImage;
+
+					if !imageStreamNode.Found() {
```

Comment:
Should have commented the code better ... I could have swore there was a comment that lead to this, but now I can't find it.  And upon revisiting, I tend to agree.  Per the conversion comment noted above, an "ImageStream" ref is caught by the existing warnings for "ImageStreamTag".

Will make updates.